src/hotspot/share/include/jvm.h

```diff
@@ -324,6 +324,15 @@ JVM_HasReferencePendingList(JNIEnv *env);
 JNIEXPORT void JNICALL
 JVM_WaitForReferencePendingList(JNIEnv *env);
 
+JNIEXPORT jboolean JNICALL
+JVM_ReferenceRefersTo(JNIEnv *env, jobject ref, jobject o);
+
+/*
+ * java.lang.ref.PhantomReference
+ */
+JNIEXPORT jboolean JNICALL
+JVM_PhantomReferenceRefersTo(JNIEnv *env, jobject ref, jobject o);
```

Comment:
Instead of a separate `PhantomReference::refersTo0` native method (and new PhantomReference.c), an alternative approach could be to detect if `ref` is an instance of `PhantomReference`  or other `Reference` and get the referent accordingly.